> What error you get without this fix?

**First I have try to install from github master:**

`npm install github:WhiskeySockets/Baileys#master`

then get:
 **triggers preinstall -> prepare -> tsc -P tsconfig.build.json**

`tsc` fails with `TS2322` in `src/Utils/noise-handler.ts` (lines ~126-128):
Type 'Buffer<ArrayBufferLike>' is not assignable to type 'Buffer<ArrayBuffer>'.
  Type 'ArrayBufferLike' is not assignable to type 'ArrayBuffer'.
    Type 'SharedArrayBuffer' is not assignable to type 'ArrayBuffer'.
      Types of property '[Symbol.toStringTag]' are incompatible.
        Type '"SharedArrayBuffer"' is not assignable to type '"ArrayBuffer"'.

Environment: Node 20/24, TypeScript 5.8.2, @types/node 20.9 (from devDependencies). The fix normalizes the HKDF outputs to Buffer<ArrayBuffer> so the Baileys build passes.

**After that I have try to build baileys library from source:**
`npm run build`

also get same error

---
# install log:
npm i "github:whiskeysockets/baileys"
npm error code 2
npm error git dep preparation failed
npm error command F:\Program Files\nodejs\node.exe F:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js install --force --cache=F:\npm-cache --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit 
--include=dev --include=peer --include=optional --no-package-lock-only --no-dry-run
npm error > baileys@7.0.0-rc.9 preinstall
npm error > node ./engine-requirements.js
npm error
npm error
npm error > baileys@7.0.0-rc.9 prepare
npm error > npm run build
npm error
npm error
npm error > baileys@7.0.0-rc.9 build
npm error > tsc -P tsconfig.build.json && tsc-esm-fix --tsconfig=tsconfig.build.json --ext=.js
npm error
npm error src/Utils/noise-handler.ts(126,3): error TS2322: Type 'Buffer<ArrayBufferLike>' is not assignable to type 'Buffer<ArrayBuffer>'.
npm error   Type 'ArrayBufferLike' is not assignable to type 'ArrayBuffer'.
npm error     Type 'SharedArrayBuffer' is not assignable to type 'ArrayBuffer'.
npm error       Types of property '[Symbol.toStringTag]' are incompatible.
npm error         Type '"SharedArrayBuffer"' is not assignable to type '"ArrayBuffer"'.
npm error src/Utils/noise-handler.ts(127,3): error TS2322: Type 'Buffer<ArrayBufferLike>' is not assignable to type 'Buffer<ArrayBuffer>'.
npm error   Type 'ArrayBufferLike' is not assignable to type 'ArrayBuffer'.
npm error     Type 'SharedArrayBuffer' is not assignable to type 'ArrayBuffer'.
npm error       Types of property '[Symbol.toStringTag]' are incompatible.
npm error         Type '"SharedArrayBuffer"' is not assignable to type '"ArrayBuffer"'.
npm error src/Utils/noise-handler.ts(128,3): error TS2322: Type 'Buffer<ArrayBufferLike>' is not assignable to type 'Buffer<ArrayBuffer>'.
npm error   Type 'ArrayBufferLike' is not assignable to type 'ArrayBuffer'.
npm error     Type 'SharedArrayBuffer' is not assignable to type 'ArrayBuffer'.
npm error       Types of property '[Symbol.toStringTag]' are incompatible.
npm error         Type '"SharedArrayBuffer"' is not assignable to type '"ArrayBuffer"'.
npm error npm warn using --force Recommended protections disabled.
npm error npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm error npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm error npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm error npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm error npm warn deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm error npm warn deprecated node-domexception@1.0.0: Use your platform's native DOMException instead
npm error npm warn using --force Recommended protections disabled.
npm error npm error code 2
npm error npm error path F:\npm-cache\_cacache\tmp\git-clonewklVf1
npm error npm error command failed
npm error npm error command C:\WINDOWS\system32\cmd.exe /d /s /c npm run build
npm error npm error A complete log of this run can be found in: F:\npm-cache\_logs\2026-01-21T01_52_38_920Z-debug-0.log 
npm error A complete log of this run can be found in: F:\npm-cache\_logs\2026-01-21T01_52_08_119Z-debug-0.log

# build log:
npm run build
> baileys@7.0.0-rc.9 build               
> tsc -P tsconfig.build.json && tsc-esm-f
ext=.js                                  
                                         
src/Utils/noise-handler.ts:126:3 - error 
>' is not assignable to type 'Buffer<Arra
  Type 'ArrayBufferLike' is not assignabl
    Type 'SharedArrayBuffer' is not assig
      Types of property '[Symbol.toString
        Type '"SharedArrayBuffer"' is not
                                         
126   salt = write! as Buffer            
      ~~~~                               
                                         
src/Utils/noise-handler.ts:127:3 - error 
>' is not assignable to type 'Buffer<Arra
  Type 'ArrayBufferLike' is not assignabl
    Type 'SharedArrayBuffer' is not assig
      Types of property '[Symbol.toString
        Type '"SharedArrayBuffer"' is not
                                         
127   encKey = read! as Buffer           
      ~~~~~~                             
                                         
src/Utils/noise-handler.ts:128:3 - error 
>' is not assignable to type 'Buffer<Arra
  Type 'ArrayBufferLike' is not assignabl
      Types of property '[Symbol.toString
        Type '"SharedArrayBuffer"' is not
                                         
128   decKey = read! as Buffer           
      ~~~~~~                             
                                         
                                         
Found 3 errors in the same file, starting

--- a/src/Utils/noise-handler.ts
+++ b/src/Utils/noise-handler.ts
@@ -64,9 +64,9 @@ export const makeNoiseHandler = ({
 
 	const data = Buffer.from(NOISE_MODE)
 	let hash = data.byteLength === 32 ? data : sha256(data)
-	let salt = hash
-	let encKey = hash
-	let decKey = hash
+	let salt: Buffer = hash
+	let encKey: Buffer = hash
+	let decKey: Buffer = hash
 	let counter = 0
 	let sentIntro = false
 
@@ -118,21 +118,21 @@ export const makeNoiseHandler = ({
 
 	const localHKDF = async (data: Uint8Array): Promise<[Buffer, Buffer]> => {
 		const key = await hkdf(Buffer.from(data), 64, { salt, info: '' })
-		return [key.subarray(0, 32) as Buffer, key.subarray(32) as Buffer]
+		return [key.subarray(0, 32), key.subarray(32)]
 	}
 
 	const mixIntoKey = async (data: Uint8Array) => {
 		const [write, read] = await localHKDF(data)
-		salt = Buffer.from(write!) as unknown as Buffer<ArrayBuffer>
-		encKey = Buffer.from(read!) as unknown as Buffer<ArrayBuffer>
-		decKey = Buffer.from(read!) as unknown as Buffer<ArrayBuffer>
+		salt = write
+		encKey = read
+		decKey = read
 		counter = 0
 	}
 
 	const finishInit = async () => {
 		isWaitingForTransport = true
 		const [write, read] = await localHKDF(new Uint8Array(0))
-		transport = new TransportState(write!, read!)
+		transport = new TransportState(write, read)
 		isWaitingForTransport = false
 
 		logger.trace('Noise handler transitioned to Transport state')

--- a/src/Utils/noise-handler.ts
+++ b/src/Utils/noise-handler.ts
@@ -9,7 +9,7 @@ import type { ILogger } from './logger'
 
 const IV_LENGTH = 12
 
-const EMPTY_BUFFER = Buffer.alloc(0) as Buffer
+const EMPTY_BUFFER = Buffer.alloc(0)
 
 const generateIV = (counter: number): Uint8Array => {
 	const iv = new ArrayBuffer(IV_LENGTH)
@@ -70,7 +70,7 @@ export const makeNoiseHandler = ({
 	let counter = 0
 	let sentIntro = false
 
-	let inBytes: Buffer = Buffer.alloc(0) as Buffer
+	let inBytes: Buffer = Buffer.alloc(0)
 
 	let transport: TransportState | null = null
 	let isWaitingForTransport = false
@@ -78,7 +78,7 @@ export const makeNoiseHandler = ({
 
 	let introHeader: Buffer
 	if (routingInfo) {
-		introHeader = Buffer.alloc(7 + routingInfo.byteLength + NOISE_HEADER.length) as Buffer
+		introHeader = Buffer.alloc(7 + routingInfo.byteLength + NOISE_HEADER.length)
 		introHeader.write('ED', 0, 'utf8')
 		introHeader.writeUint8(0, 2)
 		introHeader.writeUint8(1, 3)
@@ -87,7 +87,7 @@ export const makeNoiseHandler = ({
 		introHeader.set(routingInfo, 7)
 		introHeader.set(NOISE_HEADER, 7 + routingInfo.byteLength)
 	} else {
-		introHeader = Buffer.from(NOISE_HEADER) as Buffer
+		introHeader = Buffer.from(NOISE_HEADER)
 	}
 
 	const authenticate = (data: Uint8Array) => {
@@ -234,7 +234,7 @@ export const makeNoiseHandler = ({
 
 			const dataLen = data.byteLength
 			const introSize = sentIntro ? 0 : introHeader.length
-			const frame = Buffer.allocUnsafe(introSize + 3 + dataLen) as Buffer
+			const frame = Buffer.allocUnsafe(introSize + 3 + dataLen)
 
 			if (!sentIntro) {
 				frame.set(introHeader)

